build-system/tasks/helpers.js

```diff
@@ -412,6 +414,7 @@ function compileUnminifiedJs(srcDir, srcFilename, destDir, options) {
         )
         .pipe(source(srcFilename))
         .pipe(buffer())
+        .pipe(gulpIf(argv.coverage, istanbul()))
```

Comment:
What does this do? Are we missing somewhere that writes the report?

Reply:
Would like to address this despite @rcebulko approving to unblock.

Based on https://www.npmjs.com/package/gulp-istanbul#nodejs-testing there seems to be several steps missing, unless we are already performing them somewhere. Should I see a change to coverage reports?

Reply:
Yes, this PR doesn't contain all the steps. It merely adds instrumentation for the runtime that's served by our `gulp` server. As mentioned in the PR description, we're still looking for a way to measure and report the coverage from within a browser when our integration tests are executed.

The plan is to reach out to someone I met during NodeJS interactive who is an expert on code coverage tooling.

Reply:
For browser testing, according to https://www.npmjs.com/package/gulp-istanbul#browser-testing, we need to force `require` to return the covered files, write to a temp dir and have the browser load that, or use a karma based instanbul plugin. Aren't we already using `karma-coverage-istanbul-reporter`? I'm assume it covers our runtime code already, based on the fact that our code cov reports include `src/` among other dirs.

By adding `gulp-instanbul`, things seem to be less clear. I'm not sure what use this PR has, or maybe I'm missing something. Would it make sense to hold off until we know what the next steps are?

Reply:
Just looked at the browser testing section in https://www.npmjs.com/package/gulp-istanbul#browser-testing. It works for unit tests, but not integration tests. Let me try to summarize why:

**Unit tests:**
- During normal (non-coverage) unit test runs, we spin up a Karma server, which transforms all the test code and AMP source code with `browserify`, and runs tests inside a browser.
- During coverage runs, we instrument all AMP source code (but not test code) using `istanbul` and then run the tests as described above. Once tests are run, `karma-coverage-istanbul-reporter` generates a pretty report that we either upload to codecov.io (Travis) or display in a browser (Local).

https://github.com/ampproject/amphtml/blob/fe1fb520f20cba637356f9936b284cc60781de1a/build-system/tasks/runtime-test/runtime-test-base.js#L228-L240

**Integration tests:**
- The main difference here is that the integration test code runs inside the `gulp integration` process, but the minified runtime JS code (output of `gulp dist`) is fetched by test webpages and executed inside a Chrome (or Firefox, or Safari) process.
- There are ways to extract the coverage data for those runtime files, but it's not as simple as just using `karma-coverage-istanbul-reporter`.
- This PR merely adds a way to instrument the runtime code. I spoke to @bcoe during NodeJS Interactive, and he said he'd try to help us out with this once we have the runtime code instrumented. (Tracking issue: #22682)

This is my understanding of the current state we're in. Hope this makes sense.

Reply:
👋 @rsimha I'll try to carve out some time to look at this over the weekend, or early next week.

Reply:
Thanks Ben! I still owe you concise instructions for how to run one of our integration tests. Will do some testing on my side and send them over. Stay tuned.

Reply:
@estherkim okay to merge as-is given the above context?